### PR TITLE
Clarify Vulkan-specific items

### PR DESCRIPTION
**Pull request by MarkY-LunarG**
Since GFXR started as a Vulkan project, many of the items have no clear identifier in their name.  But now that GFXR supports DX and may support other APIs in the future, we should clarify Vulkan-specific items.

### CODE REVIEW

File: framework/encode/parameter_encoder.h [hunk 1]
```diff
@@ -78,7 +78,7 @@ class ParameterEncoder
     void EncodeFunctionPtr(T value)                                                                                   { EncodeValue(reinterpret_cast<format::AddressEncodeType>(value)); }
 
     template<typename Wrapper>
-    void EncodeHandleValue(typename Wrapper::HandleType value)                                                        { EncodeHandleIdValue(GetWrappedId<Wrapper>(value)); }
+    void EncodeHandleValue(typename Wrapper::HandleType value)                                                        { EncodeHandleIdValue(GetVulkanWrappedId<Wrapper>(value)); }
```
Comment (davidd-lunarg) [inline framework/encode/parameter_encoder.h:81]: Add `Vulkan` to the parameter encoder name as well? e.g., `EncodeHandleValue` -> `EncodeVulkanHandleValue`

File: framework/encode/custom_vulkan_api_call_encoders.cpp [hunk 2]
```diff
@@ -164,7 +164,7 @@ static void EncodeDescriptorUpdateTemplateInfo(VulkanCaptureManager*     manager
             {
                 size_t              offset = entry_info.offset + (entry_info.stride * i);
                 const VkBufferView* entry  = reinterpret_cast<const VkBufferView*>(bytes + offset);
-                encoder->EncodeHandleValue<BufferViewWrapper>(*entry);
+                encoder->EncodeHandleValue<vulkan_wrappers::BufferViewWrapper>(*entry);
```
Comment (davidd-lunarg) [inline framework/encode/custom_vulkan_api_call_encoders.cpp:167]: I wonder if there is another solution for delineating these Vulkan wrappers that doesn't require the `vulkan_wrappers::` prefix on every object. Some potential ideas:
1. Add a Vulkan-specific namespace (e.g., `gfxrecon::encode::vulkan` or `gfxrecon::encode::vk`) for all Vulkan-specific encode files. A Vulkan-specific namespace could also be added for `gfxrecon::decode`, e.g., `gfxrecon::decode::vulkan`
1. Adopt the DX12 wrapper naming pattern `[DX12 object name]_Wrapper`, so `vulkan_wrappers::BufferViewWrapper` becomes `VkBufferView_Wrapper`

Comment (MarkY-LunarG) [reply framework/encode/custom_vulkan_api_call_encoders.cpp:167]: Your idea of the full name might work, but doing additional namespaces was much more involved than I wanted to go down.

**Review by bradgrantham-lunarg — APPROVED** [review]
I'll merge when CI succeeds

**Pull request merged**